test/test_gather.cpp

```diff
@@ -826,6 +826,50 @@ TEST_F(IndexingOpTest, TakeAlongAxisIntermediateTensorReduction3_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
 }
 
+// Similar to TakeAlongAxisIntermediateTensorReduction2, but no
+// explicit squeeze of the consumer ID of the indexed domain. Still
+// segmented as a reduction of a broadcast ID is always translated to
+// a squeeze op.
+TEST_F(IndexingOpTest, TakeAlongAxisIntermediateTensorReduction4_CUDA) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape_before_gather({100, 100});
+  std::vector<int64_t> shape_after_gather({shape_before_gather[0]});
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = makeSymbolicTensor(1, DataType::Int);
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, IrBuilder::create<Double>(1));
+  auto tv3 = broadcast(tv1, {false, true});
+  auto tv4 = take_along_axis(tv2, tv3, 1);
+  auto tv5 = sum(tv4, {0, 1});
```

Comment:
IIRC, `sum` will be transformed into `squeeze` then `sum`, but what we want to test is `sum` then `squeeze` is not segmented.

Reply:
Yes, I realized I pushed this commit, and now testing an updated test, but I may have found a bug...

Reply:
Updated `TakeAlongAxisIntermediateTensorReduction4`. It's not segmented.

Found one (likely) predication bug, which is currently worked around by inserting a `set` op. I'm pretty sure it's a generic bug, not specific to this PR. Will work on it later.

Reply:
Filed an issue #282.